azure_ai_search_rag.ipynb

```diff
@@ -0,0 +1,908 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "embed_model = \"mistral-embed\"\n",
+    "\n",
+    "embeds = mistral.embeddings(\n",
+    "    model=embed_model, input=[\"this is a test\"]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can view the dimensionality of our returned embeddings, which we'll need soon when initializing our vector index:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "1024"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dims = len(embeds.data[0].embedding)\n",
+    "dims"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Index Embeddings into Azure AI Search\n",
+    "\n",
+    "Now we create our vector DB to store our vectors. For this, we need to set up an [Azure AI Search service](https://portal.azure.com/#create/Microsoft.Search).\n",
+    "\n",
+    "There are two ways to authenticate to Azure AI Search:\n",
+    "\n",
+    "1. **Service Key**: The service key can be found in the \"Settings -> Keys\" section in the left navbar of the Azure portal dashboard. Make sure to select the ADMIN key.\n",
+    "2. **Managed Identity**: This is a more secure and recommended way to authenticate. You can follow the instructions in the [official Microsoft documentation](https://learn.microsoft.com/en-us/azure/search/search-howto-managed-identities-data-sources?tabs=portal-sys%2Cportal-user) to set up Managed Identity.\n",
+    "\n",
+    "For more detailed instructions on creating an Azure AI Search service, please refer to the [official Microsoft documentation](https://learn.microsoft.com/en-us/azure/search/search-create-service-portal).\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Authenticate into Azure AI Search"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from getpass import getpass\n",
+    "from azure.identity import DefaultAzureCredential\n",
+    "from azure.core.credentials import AzureKeyCredential\n",
+    "import os\n",
+    "\n",
+    "SEARCH_SERVICE_ENDPOINT = os.getenv(\"SEARCH_SERVICE_ENDPOINT\") or getpass(\"Enter your Azure AI Search Service Endpoint: \")\n",
+    "SEARCH_SERVICE_API_KEY = os.getenv(\"SEARCH_SERVICE_API_KEY\") or getpass(\"Enter your Azure AI Search Service API Key: \")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 72,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Using AAD for authentication.\n"
+     ]
+    }
+   ],
+   "source": [
+    "USE_AAD_FOR_SEARCH = True\n",
```

Comment:
Why ask for api key in the previous cell if you set AAD auth to true here? Maybe you can move the other line to the `else` branch below and avoid asking if you don't need a key.